File: backend/local-api-module/src/main/java/org/vyatsu/localApiModule/entity/purchase/Ad.java
package org.vyatsu.localApiModule.entity.purchase;

import org.vyatsu.localApiModule.entity.item.Item;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;
import java.time.LocalDateTime;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Builder
@Data
@Entity
@Table(name = "ads")
public class Ad {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private Long id;

    @Column(name = "estimation")
    private int estimation;

    @Column(name = "link", nullable = false)
    private String link;

    @Column(name = "created_at", columnDefinition = "TIMESTAMP NOT NULL")
    private LocalDateTime createdAt;

    @ManyToOne
    @JoinColumn(name = "id_cost", nullable = false)
    private Cost cost;

    @ManyToOne
    @JoinColumn(name = "id_item", nullable = false)
    private Item item;
}

File: app/src/main/java/com/bytebusters/android/mpsystem/entity/purchase/Ad.kt
package com.bytebusters.android.mpsystem.entity.purchase

import com.bytebusters.android.mpsystem.entity.item.Item
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne
import jakarta.persistence.Table
import lombok.AllArgsConstructor
import lombok.Builder
import lombok.Data
import lombok.Getter
import lombok.NoArgsConstructor
import lombok.Setter
import java.time.LocalDateTime

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Builder
@Data
@Entity
@Table(name = "ads")
class Ad {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private val id: Long? = null

    @Column(name = "estimation")
    private val estimation = 0

    @Column(name = "link", nullable = false)
    private val link: String? = null

    @Column(name = "created_at", columnDefinition = "TIMESTAMP NOT NULL")
    private val createdAt: LocalDateTime? = null

    @ManyToOne
    @JoinColumn(name = "id_cost", nullable = false)
    private val cost: Cost? = null

    @ManyToOne
    @JoinColumn(name = "id_item", nullable = false)
    private val item: Item? = null
}
